File: platforms/wpf/wpf_sample_net/entities/Unit.cs
using System;
using System.Collections.Generic;

using wpf_lib.lib;
using wpf_lib.lib.attributes;

using wpf_generated.functions;

namespace wpf_generated.entities {

  public enum NumberOfBathroomsEnum {
    [Label("Half")]
    Half,
    [Label("1")]
    One,
    [Label("1.5")]
    OneAndHalf,
    [Label("2")]
    Two,
    [Label("3")]
    Three,
    [Label("4+")]
    FourPlus,
  }


  public class Unit : EntityBase {

    // Regular Attributes
    private string _number;
    public string Number {
      get { return _number; }
      set {
        _number = value;
        RaisePropertyChanged(nameof(Number));
      }
    }
    private double? _squareFeet;
    public double? SquareFeet {
      get { return _squareFeet; }
      set {
        _squareFeet = value;
        RaisePropertyChanged(nameof(SquareFeet));
      }
    }
    private int? _numberOfBedrooms;
    public int? NumberOfBedrooms {
      get { return _numberOfBedrooms; }
      set {
        _numberOfBedrooms = value;
        RaisePropertyChanged(nameof(NumberOfBedrooms));
      }
    }
    private NumberOfBathroomsEnum? _numberOfBathrooms;
    public NumberOfBathroomsEnum? NumberOfBathrooms {
      get { return _numberOfBathrooms; }
      set {
        _numberOfBathrooms = value;
        RaisePropertyChanged(nameof(NumberOfBathrooms));
      }
    }
    private bool _hasBalcony;
    public bool HasBalcony {
      get { return _hasBalcony; }
      set {
        _hasBalcony = value;
        RaisePropertyChanged(nameof(HasBalcony));
      }
    }

    // Derived Attributes

    // Associations

    public override string ToString() {
      return Number?.ToString();
    }

    public static Unit Create() {
      return new Unit {
        NumberOfBedrooms = 2,
        NumberOfBathrooms = NumberOfBathroomsEnum.One,
        HasBalcony = false,
      };
    }
  }
}

File: platforms/ts-chakra/src/x10_generated/small/entities/Unit.ts
import { v4 as uuid } from 'uuid';

import { addError, type FormError } from 'react_lib/form/FormProvider';
import isBlank from 'react_lib/utils/isBlank';

import { NumberOfBathroomsEnum } from '__generated__/graphql';


// Type Definition
export type Unit = {
  readonly id: string,
  readonly number: string,
  readonly squareFeet: number | null | undefined,
  readonly numberOfBedrooms: number | null | undefined,
  readonly numberOfBathrooms: NumberOfBathroomsEnum | null | undefined,
  readonly hasBalcony: boolean,
};


// Enums
export const NumberOfBathroomsEnumPairs: {
  value: NumberOfBathroomsEnum,
  label: string
}[] = [
  {
    value: NumberOfBathroomsEnum.Half,
    label: 'Half',
  },
  {
    value: NumberOfBathroomsEnum.One,
    label: '1',
  },
  {
    value: NumberOfBathroomsEnum.OneAndHalf,
    label: '1.5',
  },
  {
    value: NumberOfBathroomsEnum.Two,
    label: '2',
  },
  {
    value: NumberOfBathroomsEnum.Three,
    label: '3',
  },
  {
    value: NumberOfBathroomsEnum.FourPlus,
    label: '4+',
  },
];



// Create Default Function
export function createDefaultUnit(): Unit {
  return {
    id: uuid(),
    number: '',
    squareFeet: undefined,
    numberOfBedrooms: 2,
    numberOfBathrooms: NumberOfBathroomsEnum.One,
    hasBalcony: false,
  };
}


// Validations
export function unitCalculateErrors(unit: Unit, prefix?: string, inListIndex?: number): FormError[] {
  const errors: FormError[] = [];
  if (unit == null ) return errors;

  if (isBlank(unit.number))
    addError(errors, 'Number is required', ['number'], prefix, inListIndex);
  if (isBlank(unit.numberOfBedrooms))
    addError(errors, 'Number Of Bedrooms is required', ['numberOfBedrooms'], prefix, inListIndex);
  if (isBlank(unit.numberOfBathrooms))
    addError(errors, 'Number Of Bathrooms is required', ['numberOfBathrooms'], prefix, inListIndex);

  return errors;
}
